### PR TITLE
Android setting engine

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
## Change list

- `io.appium.java_client.AppiumSetting` became deprecated. It was moved to `io.appium.java_client.android.Setting`.

- the method `io.appium.java_client.AppiumDriver#getSettings()` was completely moved to `io.appium.java_client.android.AndroidDriver`.

- the new interface `io.appium.java_client.android.HasSettings` was added. It has default implementation. And this interface is implemented by AndroidDriver.
Methods:
  - `void setSetting(Setting setting, Object value) `
  - `JsonObject getSettings() `
  - `void ignoreUnimportantViews(Boolean compress)`
  - `void configuratorSetWaitForIdleTimeout(int timeout) `
  - `void configuratorSetWaitForSelectorTimeout(int timeout) `
  - `void configuratorSetScrollAcknowledgmentTimeout(int timeout) `
  - `void configuratorSetKeyInjectionDelay(int delay)`
  - `void configuratorSetActionAcknowledgmentTimeout(int timeout)`
This change was proposed at #410 by @truebit 
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#471

### CODE REVIEW

**Comment by TikhomirovSergey** [comment]
@SrinivasanTarget @truebit 
Could you review this PR?

**Comment by truebit** [comment]
the settings api has been changed to top level as requested by @jlipps in https://github.com/appium/appium-android-driver/pull/153
you could find the doc in https://github.com/truebit/appium/blob/master/docs/en/advanced-concepts/settings.md#supported-settings

So these settings api requests should be the same as `ignoreUnimportantViews` in top level instead of `configurator/xxxTimeout`

**Comment by TikhomirovSergey** [comment]
@truebit 
Yes and proposed code actualy does that:
https://github.com/appium/java-client/pull/477/files#diff-7daaa6d2742d959db92558292282ea56R41
https://github.com/appium/java-client/pull/477/files#diff-7daaa6d2742d959db92558292282ea56R78
https://github.com/appium/java-client/pull/477/files#diff-cebdf3eaee01b491314c7101b3103f7bR298
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/MobileCommand.java#L92

Isn't it?

File: src/main/java/io/appium/java_client/android/Setting.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+package io.appium.java_client.android;
+
+/**
+ * Enums defining constants for Appium Settings which can be set and toggled during a test session.
+ */
+public enum Setting {
+
+    IGNORE_UNIMPORTANT_VIEWS("ignoreUnimportantViews"),
+    CONFIGURATOR("configurator");
```
Comment (truebit) [inline src/main/java/io/appium/java_client/android/Setting.java]: no, this code is still the two level configurator. Now it should be like `ignoreUnimportantViews`. Every setting here for configurators should be individivual

**Comment by truebit** [comment]
all below settings are at same level like `ignoreUnimportantViews`. there is no 'configurator' sub setting
> "actionAcknowledgmentTimeout" - Int which is the same as setActionAcknowledgmentTimeout. If a negative value is given, it would set to default(3 * 1000 milliseconds)
> 
> "keyInjectionDelay" - Int which is the same as setKeyInjectionDelay. If a negative value is given, it would set to default(0 milliseconds)
> 
> "scrollAcknowledgmentTimeout" - Int which is the same as setScrollAcknowledgmentTimeout. If a negative value is given, it would set to default(200 milliseconds)
> 
> "waitForIdleTimeout" - Int which is the same as setWaitForIdleTimeout. If a negative value is given, it would set to default(10 * 1000 milliseconds)
> 
> "waitForSelectorTimeout" - Int which is the same as setWaitForSelectorTimeout. If a negative value is given, it would set to default(10 * 1000 milliseconds)

**Comment by TikhomirovSergey** [comment]
Ok. Will iptrove this a bit later.

**Pull request merged**